### DISCUSSION TITLE
Some daily counters does not reset perfectly at midnight

### DISCUSSION THREAD

**Discussion by Claudio1L** [wills106/homeassistant-solax-modbus#301] [Q&A]
Hi, in my x1-retrofit G3 I'm experiencing some issue with the daily reset of some counters. I think was there also before, but now I'm using them for some calculation, and this wrong reset cause some issue to some helpers and templates referring to them.
Any chance to fix them?

![immagine](https://user-images.githubusercontent.com/97904190/213814630-17132ffa-1576-43bd-9f25-91f93296444b.png)
![immagine](https://user-images.githubusercontent.com/97904190/213814644-8feeac0d-beed-460d-825f-d284b9377bc5.png)
![immagine](https://user-images.githubusercontent.com/97904190/213814663-b87b89bb-3d83-41a0-ba53-9a219397b4eb.png)
![immagine](https://user-images.githubusercontent.com/97904190/213814677-3cad4a8d-8225-4f59-b538-6aabbf124beb.png)

**Comment by wills106** [comment]
Have you made sure your Inverter RTC is matching your HA installation? The RTC on the Inverters does drift out over time.
Also if you have a relatively slow poll rate you could be having the update from the Inverter come through at different times.

**Comment by Claudio1L** [comment]
Polling rate Is 5 seconds, should I reduce it? What’s the minimum recommended?

Thanks

**Comment by Claudio1L** [comment]
It seems poling rtc every 5 sec
￼

**Reply by wills106** [reply]
I would leave your polling how it is, I wouldn't decrease it.

Potentially increasing your poll rate might make them come through at the same time. But you might also stagger them further. Also with a slower speed you you lose the resolution of generation / consumption.

I noticed mine come through at slightly different times.

I don't think there is much I can do within the Integration. I want to keep the Integration pure that it only outputs what the Inverter outputs.
I don't know if you can do anything by templating the sensor for a time condition? Or the utility sensor?

**Comment by Claudio1L** [comment]
I'm using the following template:

home_energy_today:
    friendly_name: 'Home Energy Today'
    unique_id: sensor.home_energy_today
    device_class: energy
    value_template: "{{ ((states('sensor.produzione_giornaliera')|float) 
                       + (states('sensor.solax_today_s_import_energy')|float)
                       + (states('sensor.solax_today_s_yield')|float)
                       - (states('sensor.solax_e_charge_today')|float)
                       - (states('sensor.solax_today_s_export_energy')|float))|round(3)}}"
    unit_of_measurement: "kWh"

at midnight, there is always a behavior similar to the follow, with spike sometime  higher than this, disturbing other “utility meter” helpers referring to this for montly and yearly calculation.

![immagine](https://user-images.githubusercontent.com/97904190/213879044-7101afd9-ff55-431e-a41e-ddfa6bbfe0e8.png)

**Reply by Claudio1L** [reply]
I should find a way to avoid at least the spike, but no way till now

**Reply by Paiklabs** [reply]
Hi, I have exactly the same problem. In the native Energy dashboard HA as well as in Grafana, the daily totals show nonsense. The measurement from the previous day is taken into account.
